# Copyright (c) 2006, 2007
# by Andreas Untergasser and Harm Nijveen
# All rights reserved.
#
# See COPYING-GPL3 for full GPLv3 license.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
# GNU General Public License for more details.
#
# # You should have received a copy of the
# GNU General Public License along with this program.
# If not, see <https://www.gnu.org/licenses/>.


st_help = [
    {"header": {"body": "Cautions", "anchor": "cautions"}},
    {"markdown": {"body": "Some of the most important issues in primer picking can be addressed only before using Primer3. These are sequence quality (including making sure the sequence is not vector and not chimeric) and avoiding repetitive elements.\nTechniques for avoiding problems include a thorough understanding of possible vector contaminants and cloning artifacts coupled with database searches using blast, fasta, or other similarity searching program to screen for vector contaminants and possible repeats. Repbase (J. Jurka, A.F.A. Smit, C. Pethiyagoda, and others, 1995-1996) [ftp://ftp.ncbi.nih.gov/repository/repbase](ftp://ftp.ncbi.nih.gov/repository/repbase) ) is an excellent source of repeat sequences and pointers to the literature. Primer3 now allows you to screen candidate oligos against a Mispriming Library (or a Mishyb Library in the case of internal oligos).\nSequence quality can be controlled by manual trace viewing and quality clipping or automatic quality clipping programs. Low-quality bases should be changed to N's or can be made part of Excluded Regions. The beginning of a sequencing read is often problematic because of primer peaks, and the end of the read often contains many low-quality or even meaningless called bases. Therefore when picking primers from single-pass sequence it is often best to use the Included Region parameter to ensure that Primer3 chooses primers in the high quality region of the read.\nIn addition, Primer3 takes as input a [Sequence Quality](#SEQUENCE_QUALITY) list for use with those base calling programs such as Phred that output this information."}},
    {"header": {"body": "Script Input Parameters", "anchor": "INPUT_PARAMETERS"}},
    {"header": {"body": "Task", "anchor": "TASK"}},
    {"markdown": {"body": "- [Detection](#DETECTION)\n- [Cloning](#CLONING)\n[Sequencing](#SEQUENCING)\n- [Primer List](#PRIMER_LIST)\n- [Primer Check](#PRIMER_CHECK)."}},
    {"subheader": {"body": "Task: Detection", "anchor": "DETECTION"}},
    {"markdown": {"body": "The Detection task can be used for designing standard PCR primers or hybridisation oligos to DETECT a given sequence.\nThe user can indicate:\nexcluded regions - primers are not allowed to bind in this region\ntargets - primers must amplify one of the targets\nincluded region - primers must bind within this region\nPrimer3Plus will select the primer pair which fits best to all selected parameters. It can be located anywhere in the sequence, only limited by the regions and targets described above."}},
    {"subheader": {"body": "Task: Cloning", "anchor": "CLONING"}},
    {"markdown": {"body": "The Cloning task can be used to design primers ending or starting exactly at the boundary of the included region.\nTo clone for example open reading frames the 5'End of the primers should be fixed. Primer3plus picks primers of various length, all forward primers starting at the same nucleotide (A from ATG) and all reverse primers starting at the same nucleotide (G from TAG). To have this functionality the parameter \"Fix the x prime end of the primer\" in general setting should be set to 5 (default).\nTo to distinguish between different alleles primers must bind with their 3'End fixed to the varying nucleotides. If the parameter \"Fix the x prime end of the primer\" in general setting is set to 3, primer3plus picks primers of various length, all primers ending at the same nucleotide.\nPrimer3Plus picks out of all primers the best pair and orders them by quality.\nATTENTION: Due to the inflexibility of the primer position, only the primer length can be altered. In many cases this leads to primers of low quality. Select the cloning function ONLY if you require your primers to start or to end at a certain nucleotide."}},
    {"subheader": {"body": "Fix the x prime end of the primer", "anchor": "FIX_PRIMER_END"}},
    {"markdown": {"body": "This parameter can be set to 5 (default) or 3. It indicates for Primer3Plus which End of the primer should be fixed and which can be extended or cut."}},
    {"subheader": {"body": "Task: Sequencing", "anchor": "SEQUENCING"}},
    {"markdown": {"body": "The Sequencing task is developed to design a series of primers on both the forward and reverse strands that can be used for custom primer-based (re-)sequencing of clones. Targets can be defined in the sequence which will be sequenced optimally. The pattern how Primer3Plus picks the primers can be modified on the Advanced settings tab."}},
    {"subheader": {"body": "Lead", "anchor": "SEQUENCING_LEAD"}},
    {"markdown": {"body": "Defines the space from the start of the primer to the point were the trace signals are readable (default 50 bp)."}},
    {"subheader": {"body": "Spacing", "anchor": "SEQUENCING_SPACING"}},
    {"markdown": {"body": "Defines the space from the start of the primer to the start of the next primer on the same strand (default 500 bp)."}},
    {"subheader": {"body": "Interval", "anchor": "SEQUENCING_INTERVAL"}},
    {"markdown": {"body": "Defines the space from the start of the primer to the start of the next primer on the opposite strand (default 250 bp)."}},
    {"subheader": {"body": "Accuracy", "anchor": "SEQUENCING_ACCURACY"}},
    {"markdown": {"body": "Defines the size of the area in which primer3plus searches for the best primer (default 20 bp)."}},
    {"subheader": {"body": "Pick Reverse Primers", "anchor": "SEQUENCING_REVERSE"}},
    {"markdown": {"body": "Select \"Pick Reverse Primers\" to pick also primers on the reverse strand (selected by default)."}},
    {"subheader": {"body": "Task: Primer List", "anchor": "PRIMER_LIST"}},
    {"markdown": {"body": "With the Primer List task all possible primers that can be designed on the target sequence and meet the current settings will be returned with their corresponding characteristics. This task basically allows manual selection of primers. The run time of the Primer List task can be relatively long, especially when lengthy target sequences are submitted."}},
    {"subheader": {"body": "Task: Primer Check", "anchor": "PRIMER_CHECK"}},
    {"markdown": {"body": "The Primer Check task can be used to obtain information on a specified primer, like its melting temperature or self complementarity. For this task no template sequence is required and only the primer sequence has to be provided."}},
    {"subheader": {"body": "Settings preset", "anchor": "SETTINGS_PRESET"}},
    {"markdown": {"body": "A collection of settings for specific applications."}},
    {"subheader": {"body": "Naming of Primers\nPrimer3Plus will create automatically a name for each primer based on the Sequence ID, the primer number and the primer acronym."}},
    {"subheader": {"body": "Left Primer Acronym", "anchor": "PRIMER_NAME_ACRONYM_LEFT"}},
    {"markdown": {"body": "Acronym for the left primer, by default ."}},
    {"subheader": {"body": "Internal Oligo Acronym", "anchor": "PRIMER_NAME_ACRONYM_INTERNAL_OLIGO"}},
    {"markdown": {"body": "Acronym for the internal oligo, by default I."}},
    {"subheader": {"body": "Right Primer Acronym", "anchor": "PRIMER_NAME_ACRONYM_RIGHT"}},
    {"markdown": {"body": "Acronym for the right primer, by default ."}},
    {"subheader": {"body": "Primer Name Spacer", "anchor": "PRIMER_NAME_ACRONYM_SPACER"}},
    {"markdown": {"body": "Spacer primer3plus uses between name, number and acronym, by default ."}},
    {"header": {"body": "Input Parameters", "anchor": "INPUT_PARAMETERS"}},
    {"header": {"body": "Source Sequence", "anchor": "SEQUENCE"}},
    {"markdown": {"body": "The sequence from which to select primers or hybridization oligos."}},
    {"subheader": {"body": "Sequence Id", "anchor": "PRIMER_SEQUENCE_ID"}},
    {"markdown": {"body": "An identifier that is reproduced in the output to enable you to identify the chosen primers."}},
    {"subheader": {"body": "Targets", "anchor": "TARGET"}},
    {"markdown": {"body": "If one or more Targets is specified then a legal primer pair must flank at least one of them. A Target might be a simple sequence repeat site (for example a CA repeat) or a single-base-pair polymorphism. The value should be a space-separated list of\n```\nstart,length\n```\npairs where *start* is the index of the first base of a Target, and *length* is its length."}},
    {"subheader": {"body": "Excluded Regions", "anchor": "EXCLUDED_REGION"}},
    {"markdown": {"body": "Primer oligos may not overlap any region specified in this tag. The associated value must be a space-separated list of\n```\nstart,length\n```\npairs where *start* is the index of the first base of the excluded region, and *length* is its length. This tag is useful for tasks such as excluding regions of low sequence quality or for excluding regions containing repetitive elements such as ALUs or LINEs."}},
    {"subheader": {"body": "Product Size Range", "anchor": "PRIMER_PRODUCT_SIZE_RANGE"}},
    {"markdown": {"body": "A list of product size ranges, for example\n```\n150-250 100-300 301-400\n```\nPrimer3 first tries to pick primers in the first range. If that is not possible, it goes to the next range and tries again. It continues in this way until it has either picked all necessary primers or until there are no more ranges. For technical reasons this option makes much lighter computational demands than the Product Size option."}},
    {"subheader": {"body": "Product Size", "anchor": "PRIMER_PRODUCT_SIZE"}},
    {"markdown": {"body": "Minimum, Optimum, and Maximum lengths (in bases) of the PCR product. Primer3 will not generate primers with products shorter than Min or longer than Max, and with default arguments Primer3 will attempt to pick primers producing products close to the Optimum length."}},
    {"subheader": {"body": "Number To Return", "anchor": "PRIMER_NUM_RETURN"}},
    {"markdown": {"body": "The maximum number of primer pairs to return. Primer pairs returned are sorted by their \"quality\", in other words by the value of the objective function (where a lower number indicates a better primer pair). Caution: setting this parameter to a large value will increase running time."}},
    {"subheader": {"body": "Max 3' Stability", "anchor": "PRIMER_MAX_END_STABILITY"}},
    {"markdown": {"body": "The maximum stability for the last five 3' bases of a left or right primer. Bigger numbers mean more stable 3' ends. The value is the maximum delta G (kcal/mol) for duplex disruption for the five 3' bases as calculated using the Nearest-Neighbor parameter values specified by the option of ['Table of thermodynamic parameters'](#PRIMER_TM_SANTALUCIA). For example if the table of thermodynamic parameters suggested by [SantaLucia 1998, DOI:10.1073/pnas.95.4.1460](http://dx.doi.org/10.1073/pnas.95.4.1460) is used the deltaG values for the most stable and for the most labile 5mer duplex are 6.86 kcal/mol (GCGCG) and 0.86 kcal/mol (TATAT) respectively. If the table of thermodynamic parameters suggested by [Breslauer et al. 1986, 10.1073/pnas.83.11.3746](http://dx.doi.org/10.1073/pnas.83.11.3746) is used the deltaG values for the most stable and for the most labile 5mer are 13.4 kcal/mol (GCGCG) and 4.6 kcal/mol (TATAC) respectively."}},
    {"subheader": {"body": "Max Mispriming", "anchor": "PRIMER_MAX_MISPRIMING"}},
    {"markdown": {"body": "The maximum allowed weighted similarity with any sequence in Mispriming Library. Default is 12."}},
    {"subheader": {"body": "Max Template Mispriming", "anchor": "PRIMER_MAX_TEMPLATE_MISPRIMING"}},
    {"markdown": {"body": "The maximum allowed similarity to ectopic sites in the sequence from which you are designing the primers. The scoring system is the same as used for Max Mispriming, except that an ambiguity code is never treated as a consensus."}},
    {"subheader": {"body": "Pair Max Mispriming", "anchor": "PRIMER_PAIR_MAX_MISPRIMING"}},
    {"markdown": {"body": "The maximum allowed sum of similarities of a primer pair (one similarity for each primer) with any single sequence in Mispriming Library. Default is 24. Library sequence weights are not used in computing the sum of similarities."}},
    {"subheader": {"body": "**Pair Max Template Mispriming**", "anchor": "PRIMER_PAIR_MAX_TEMPLATE_MISPRIMING"}},
    {"markdown": {"body": "**\nThe maximum allowed summed similarity of both primers to ectopic sites in the sequence from which you are designing the primers. The scoring system is the same as used for Max Mispriming, except that an ambiguity code is never treated as a consensus."}},
    {"subheader": {"body": "Primer Size", "anchor": "PRIMER_SIZE"}},
    {"markdown": {"body": "Minimum, Optimum, and Maximum lengths (in bases) of a primer oligo. Primer3 will not pick primers shorter than Min or longer than Max, and with default arguments will attempt to pick primers close with size close to Opt. Min cannot be smaller than 1. Max cannot be larger than 36. (This limit is governed by maximum oligo size for which melting-temperature calculations are valid.) Min cannot be greater than Max."}},
    {"subheader": {"body": "Primer T$_m$", "anchor": "PRIMER_TM"}},
    {"markdown": {"body": "Minimum, Optimum, and Maximum melting temperatures (Celsius) for a primer oligo. Primer3 will not pick oligos with temperatures smaller than Min or larger than Max, and with default conditions will try to pick primers with melting temperatures close to Opt.\nBy default Primer3 uses the oligo melting temperature formula and the table of thermodynamic parameters given in [Breslauer et al. 1986, DOI:10.1073/pnas.83.11.3746](http://dx.doi.org/10.1073/pnas.83.11.3746)~For more information see caption [Table of thermodynamic parameters](#PRIMER_TM_SANTALUCIA)."}},
    {"subheader": {"body": "Maximum T$_m$ Difference", "anchor": "PRIMER_MAX_DIFF_TM"}},
    {"markdown": {"body": "Maximum acceptable (unsigned) difference between the melting temperatures of the left and right primers."}},
    {"subheader": {"body": "Table of thermodynamic parameters", "anchor": "PRIMER_TM_SANTALUCIA"}},
    {"markdown": {"body": "Option for the table of Nearest-Neighbor thermodynamic parameters and for the method of melting temperature calculation. Two different tables of thermodynamic parameters are available:\n1.  [Breslauer et al. 1986, DOI:10.1073/ pnas.83.11.3746](http://dx.doi.org/10.1073/pnas.83.11.3746) In that case the formula for melting temperature calculation suggested by [Rychlik et al. 1990](http://www.pubmedcentral.nih.gov/articlerender.fcgi?tool=pubmed&pubmedid=2243783) is used (this is used until Primer3 version 1.0.1). This is the default value of Primer3 (for backward compatibility).\n2.  [SantaLucia 1998, DOI:10.1073/pnas.95.4.1460](http://dx.doi.org/10.1073/pnas.95.4.1460) This is the *recommended* value.\nFor specifying the salt correction method for melting temperature calculation see [Salt correction formula](#PRIMER_SALT_CORRECTIONS)."}},
    {"subheader": {"body": "Product T$_m$", "anchor": "PRIMER_PRODUCT_TM"}},
    {"markdown": {"body": "The minimum, optimum, and maximum melting temperature of the amplicon. Primer3 will not pick a product with melting temperature less than min or greater than max. If Opt is supplied and the [Penalty Weights for Product Size](#PAIR_WT_PRODUCT_TM) are non-0 Primer3 will attempt to pick an amplicon with melting temperature close to Opt.\nThe maximum allowed melting temperature of the amplicon. Primer3 calculates product T$_m$ calculated using the formula from Bolton and McCarthy, PNAS 84:1390 (1962) as presented in Sambrook, Fritsch and Maniatis, Molecular Cloning, p 11.46 (1989, CSHL Press).\n\n$T_m = 81.5 + 16.6\cdot\log_{10}([{Na}^+]) + 0.41\cdot(\%{GC}) - {600}\div{length}$,\n\nwhere [Na$^+$] is the molar sodium concentration, (%GC) is the percent of Gs and Cs in the sequence, and length is the length of the sequence.\nA similar formula is used by the prime primer selection program in GCG ( http://www.gcg.com ), which instead uses 675.0 / length in the last term (after F. Baldino, Jr, M.-F. Chesselet, and M.E. Lewis, Methods in Enzymology 168:766 (1989) eqn (1) on page 766 without the mismatch and formamide terms). The formulas here and in Baldino et al. assume Na+ rather than K+. According to J.G. Wetmur, Critical Reviews in BioChem. and Mol. Bio. 26:227 (1991) 50 mM K+ should be equivalent in these formulae to .2 M Na+. Primer3 uses the same salt concentration value for calculating both the primer melting temperature and the oligo melting temperature. If you are planning to use the PCR product for hybridization later this behavior will not give you the T$_m$ under hybridization conditions."}},
    {"subheader": {"body": "Primer GC%", "anchor": "PRIMER_GC_PERCENT"}},
    {"markdown": {"body": "Minimum, Optimum, and Maximum percentage of Gs and Cs in any primer."}},
    {"subheader": {"body": "Max Complementarity", "anchor": "PRIMER_SELF_ANY"}},
    {"markdown": {"body": "The maximum allowable local alignment score when testing a single primer for (local) self-complementarity and the maximum allowable local alignment score when testing for complementarity between left and right primers. Local self-complementarity is taken to predict the tendency of primers to anneal to each other without necessarily causing self-priming in the PCR. The scoring system gives 1.00 for complementary bases, -0.25 for a match of any base (or N) with an N, -1.00 for a mismatch, and -2.00 for a gap. Only single-base-pair gaps are allowed. For example, the alignment\n```\n    5' ATCGNA 3'\n       || | |\n    3' TA-CGT 5'\n    \n```\nis allowed (and yields a score of 1.75), but the alignment\n```\n    5' ATCCGNA 3'\n       ||  | |\n    3' TA--CGT 5'\n    \n```\nis not considered. Scores are non-negative, and a score of 0.00 indicates that there is no reasonable local alignment between two oligos."}},
    {"subheader": {"body": "Max 3' Complementarity", "anchor": "PRIMER_SELF_END"}},
    {"markdown": {"body": "The maximum allowable 3'-anchored global alignment score when testing a single primer for self-complementarity, and the maximum allowable 3'-anchored global alignment score when testing for complementarity between left and right primers. The 3'-anchored global alignment score is taken to predict the likelihood of PCR-priming primer-dimers, for example\n```\n    5' ATGCCCTAGCTTCCGGATG 3'\n                 ||| |||||\n              3' AAGTCCTACATTTAGCCTAGT 5'\n    \n```\nor\n```\n    5` AGGCTATGGGCCTCGCGA 3'\n                   ||||||\n                3' AGCGCTCCGGGTATCGGA 5'\n    \n```\nThe scoring system is as for the Max Complementarity argument. In the examples above the scores are 7.00 and 6.00 respectively. Scores are non-negative, and a score of 0.00 indicates that there is no reasonable 3'-anchored global alignment between two oligos. In order to estimate 3'-anchored global alignments for candidate primers and primer pairs, Primer assumes that the sequence from which to choose primers is presented 5'->3'. It is nonsensical to provide a larger value for this parameter than for the Maximum (local) Complementarity parameter because the score of a local alignment will always be at least as great as the score of a global alignment."}},
    {"subheader": {"body": "Max Poly-X", "anchor": "PRIMER_MAX_POLY_X"}},
    {"markdown": {"body": "The maximum allowable length of a mononucleotide repeat, for example AAAAAA."}},
    {"subheader": {"body": "Included Region", "anchor": "INCLUDED_REGION"}},
    {"markdown": {"body": "A sub-region of the given sequence in which to pick primers. For example, often the first dozen or so bases of a sequence are vector, and should be excluded from consideration. The value for this parameter has the form\n```\nstart,length\n```\nwhere *start* is the index of the first base to consider, and *length* is the number of subsequent bases in the primer-picking region."}},
    {"subheader": {"body": "Start Codon Position", "anchor": "PRIMER_START_CODON_POSITION"}},
    {"markdown": {"body": "This parameter should be considered EXPERIMENTAL at this point. Please check the output carefully; some erroneous inputs might cause an error in Primer3. Index of the first base of a start codon. This parameter allows Primer3 to select primer pairs to create in-frame amplicons e.g. to create a template for a fusion protein. Primer3 will attempt to select an in-frame left primer, ideally starting at or to the left of the start codon, or to the right if necessary. Negative values of this parameter are legal if the actual start codon is to the left of available sequence. If this parameter is non-negative Primer3 signals an error if the codon at the position specified by this parameter is not an ATG. A value less than or equal to -10^6 indicates that Primer3 should ignore this parameter.\nPrimer3 selects the position of the right primer by scanning right from the left primer for a stop codon. Ideally the right primer will end at or after the stop codon."}},
    {"subheader": {"body": "Mispriming Library", "anchor": "PRIMER_MISPRIMING_LIBRARY"}},
    {"markdown": {"body": "This selection indicates what mispriming library (if any) Primer3 should use to screen for interspersed repeats or for other sequence to avoid as a location for primers. The human and rodent libraries on the web page are adapted from Repbase (J. Jurka, A.F.A. Smit, C. Pethiyagoda, et al., 1995-1996) [ftp://ftp.ncbi.nih.gov/repository/repbase](ftp://ftp.ncbi.nih.gov/repository/repbase) . The human library is humrep.ref concatenated with simple.ref, translated to FASTA format. There are two rodent libraries. One is rodrep.ref translated to FASTA format, and the other is rodrep.ref concatenated with simple.ref, translated to FASTA format.\nThe *Drosophila* library is the concatenation of two libraries from the [Berkeley Drosophila Genome Project](http://www.fruitfly.org):\n1.  A library of transposable elements [*The transposable elements of the Drosophila melanogaster euchromatin - a genomics perspective* J.S. Kaminker, C.M. Bergman, B. Kronmiller, J. Carlson, R. Svirskas, S. Patel, E. Frise, D.A. Wheeler, S.E. Lewis, G.M. Rubin, M. Ashburner and S.E. Celniker Genome Biology (2002) 3(12):research0084.1-0084.20](http://genomebiology.com/2002/3/12/research/0084), [http://www.fruitfly.org/p_disrupt/datasets/ASHBURNER/D_mel_transposon_sequence_set.fasta](http://www.fruitfly.org/p_disrupt/datasets/ASHBURNER/D_mel_transposon_sequence_set.fasta)\n    \n3.  A library of repetitive DNA sequences [http://www.fruitfly.org/sequence/sequence_db/na_re.dros](http://www.fruitfly.org/sequence/sequence_db/na_re.dros).\nBoth were downloaded 6/23/04.\nThe contents of the libraries can be viewed at the following links:\n- [HUMAN](https://www.bioinformatics.nl/cgi-bin/primer3plus/cat_humrep_and_simple.cgi) (contains microsatellites)\n- [RODENT_AND_SIMPLE](https://www.bioinformatics.nl/cgi-bin/primer3plus/cat_rodrep_and_simple.cgi) (contains microsatellites)\n- [RODENT](https://www.bioinformatics.nl/cgi-bin/primer3plus/cat_rodent_ref.cgi) (does not contain microsatellites)\n- [DROSOPHILA](https://www.bioinformatics.nl/cgi-bin/primer3plus/cat_drosophila.cgi)."}},
    {"subheader": {"body": "CG Clamp", "anchor": "PRIMER_GC_CLAMP"}},
    {"markdown": {"body": "Require the specified number of consecutive Gs and Cs at the 3' end of both the left and right primer. (This parameter has no effect on the hybridization oligo if one is requested.."}},
    {"subheader": {"body": "Concentration of monovalent cations", "anchor": "PRIMER_SALT_CONC"}},
    {"markdown": {"body": "The millimolar concentration of salt (usually KCl) in the PCR. Primer3 uses this argument to calculate oligo melting temperatures."}},
    {"subheader": {"body": "Concentration of divalent cations", "anchor": "PRIMER_DIVALENT_CONC"}},
    {"markdown": {"body": "The millimolar concentration of divalent salt cations (usually MgCl$^2+$ in the PCR). Primer3 converts concentration of divalent cations to concentration of monovalent cations using formula suggested in the paper [Ahsen et al., 2001](http://www.clinchem.org/cgi/content/full/47/11/1956)\n\n$[{Monovalent\ cations}] = [{Monovalent\ cations}] + 120\cdot\sqrt{[{divalent\ cations}] - [{dNTP}]}$\n\nAccording to the formula concentration of desoxynucleotide triphosphate [dNTP] must be smaller than concentration of divalent cations. The concentration of dNTPs is included to the formula beacause of some magnesium is bound by the dNTP. Attained concentration of monovalent cations is used to calculate oligo/primer melting temperature. See [Concentration of dNTPs](#PRIMER_DNTP_CONC) to specify the concentration of dNTPs."}},
    {"subheader": {"body": "Concentration of dNTPs", "anchor": "PRIMER_DNTP_CONC"}},
    {"markdown": {"body": "The millimolar concentration of deoxyribonucleotide triphosphate. This argument is considered only if [Concentration of divalent cations](#PRIMER_DIVALENT_CONC) is specified."}},
    {"subheader": {"body": "Salt correction formula", "anchor": "PRIMER_SALT_CORRECTIONS"}},
    {"markdown": {"body": "Option for specifying the salt correction formula for the melting temperature calculation.\nThere are three different options available:\n1.  [Schildkraut and Lifson 1965, DOI:10.1002/bip.360030207](http://dx.doi.org/10.1002/bip.360030207) (this is used until the version 1.0.1 of Primer3).The default value of Primer3 version 1.1.0 (for backward compatibility)\n2.  [SantaLucia 1998, DOI:10.1073/pnas.95.4.1460](http://dx.doi.org/10.1073/pnas.95.4.1460) This is the *recommended* value.\n3.  [Owczarzy et al. 2004, DOI:10.1021/bi034621r](http://dx.doi.org/10.1021/bi034621r)."}},
    {"subheader": {"body": "Annealing Oligo Concentration", "anchor": "PRIMER_DNA_CONC"}},
    {"markdown": {"body": "The nanomolar concentration of annealing oligos in the PCR. Primer3 uses this argument to calculate oligo melting temperatures. The default (50nM) works well with the standard protocol used at the Whitehead/MIT Center for Genome Research--0.5 microliters of 20 micromolar concentration for each primer oligo in a 20 microliter reaction with 10 nanograms template, 0.025 units/microliter Taq polymerase in 0.1 mM each dNTP, 1.5mM MgCl2, 50mM KCl, 10mM Tris-HCL (pH 9.3) using 35 cycles with an annealing temperature of 56 degrees Celsius. This parameter corresponds to 'c' in Rychlik, Spencer and Rhoads' equation (ii) (Nucleic Acids Research, vol 18, num 21) where a suitable value (for a lower initial concentration of template) is \"empirically determined\". The value of this parameter is less than the actual concentration of oligos in the reaction because it is the concentration of annealing oligos, which in turn depends on the amount of template (including PCR product) in a given cycle. This concentration increases a great deal during a PCR; fortunately PCR seems quite robust for a variety of oligo melting temperatures."}},
    {"subheader": {"body": "Max Ns Accepted", "anchor": "PRIMER_NUM_NS_ACCEPTED"}},
    {"markdown": {"body": "Maximum number of unknown bases (N) allowable in any primer."}},
    {"subheader": {"body": "Liberal Base", "anchor": "PRIMER_LIBERAL_BASE"}},
    {"markdown": {"body": "This parameter provides a quick-and-dirty way to get Primer3 to accept IUB / IUPAC codes for ambiguous bases (i.e. by changing all unrecognized bases to N). If you wish to include an ambiguous base in an oligo, you must set [Max Ns Accepted](#PRIMER_NUM_NS_ACCEPTED) to a non-0 value. Perhaps '-' and '* ' should be squeezed out rather than changed to 'N', but currently they simply get converted to N's. The authors invite user comments."}},
    {"subheader": {"body": "First Base Index", "anchor": "PRIMER_FIRST_BASE_INDEX"}},
    {"markdown": {"body": "The index of the first base in the input sequence. For input and output using 1-based indexing (such as that used in GenBank and to which many users are accustomed) set this parameter to 1. For input and output using 0-based indexing set this parameter to 0. (This parameter also affects the indexes in the contents of the files produced when the primer file flag is set.) In the WWW interface this parameter defaults to 1."}},
    {"subheader": {"body": "Inside Target Penalty", "anchor": "PRIMER_INSIDE_PENALTY"}},
    {"markdown": {"body": "Non-default values valid only for sequences with 0 or 1 target regions. If the primer is part of a pair that spans a target and overlaps the target, then multiply this value times the number of nucleotide positions by which the primer overlaps the (unique) target to get the 'position penalty'. The effect of this parameter is to allow Primer3 to include overlap with the target as a term in the objective function."}},
    {"subheader": {"body": "Outside Target Penalty", "anchor": "PRIMER_OUTSIDE_PENALTY"}},
    {"markdown": {"body": "Non-default values valid only for sequences with 0 or 1 target regions. If the primer is part of a pair that spans a target and does not overlap the target, then multiply this value times the number of nucleotide positions from the 3' end to the (unique) target to get the 'position penalty'. The effect of this parameter is to allow Primer3 to include nearness to the target as a term in the objective function."}},
    {"subheader": {"body": "Show Debuging Info", "anchor": "SHOW_DEBUGGING"}},
    {"markdown": {"body": "Include the input to primer3_core as part of the output."}},
    {"subheader": {"body": "Lowercase masking", "anchor": "PRIMER_LOWERCASE_MASKING"}},
    {"markdown": {"body": "If checked candidate primers having lowercase letter exactly at 3' end are rejected. This option allows to design primers overlapping lowercase-masked regions. This property relies on the assumption that masked features (e.g. repeats) can partly overlap primer, but they cannot overlap the 3'-end of the primer. In other words, the lowercase letters in other positions are accepted, assuming that the masked features do not influence the primer performance if they do not overlap the 3'-end of primer."}},
    {"header": {"body": "Sequence Quality", "anchor": "PRIMER_SEQUENCE_QUALITY"}},
    {"markdown": {"body": "A list of space separated integers. There must be exactly one integer for each base in the Source Sequence if this argument is non-empty. High numbers indicate high confidence in the base call at that position and low numbers indicate low confidence in the base call at that position."}},
    {"subheader": {"body": "Min Sequence Quality", "anchor": "PRIMER_MIN_QUALITY"}},
    {"markdown": {"body": "The minimum sequence quality (as specified by Sequence Quality) allowed within a primer."}},
    {"subheader": {"body": "Min 3' Sequence Quality", "anchor": "PRIMER_MIN_END_QUALITY"}},
    {"markdown": {"body": "The minimum sequence quality (as specified by Sequence Quality) allowed within the 3' pentamer of a primer."}},
    {"subheader": {"body": "Sequence Quality Range Min", "anchor": "PRIMER_QUALITY_RANGE_MIN"}},
    {"markdown": {"body": "The minimum legal sequence quality (used for interpreting Min Sequence Quality and Min 3' Sequence Quality)."}},
    {"subheader": {"body": "Sequence Quality Range Max", "anchor": "PRIMER_QUALITY_RANGE_MAX"}},
    {"markdown": {"body": "The maximum legal sequence quality (used for interpreting Min Sequence Quality and Min 3' Sequence Quality)."}},
    {"header": {"body": "Penalty Weights", "anchor": "generic_penalty_weights"}},
    {"markdown": {"body": "This section describes \"penalty weights\", which allow the user to modify the criteria that Primer3 uses to select the \"best\" primers. There are two classes of weights: for some parameters there is a 'Lt' (less than) and a 'Gt' (greater than) weight. These are the weights that Primer3 uses when the value is less or greater than (respectively) the specified optimum. The following parameters have both 'Lt' and 'Gt' weights:\n- Product Size\n- Primer Size\n- Primer T$_m$\n- Product T$_m$\n- Primer GC%\n- Hyb Oligo Size\n- Hyb Oligo T$_m$\n- Hyb Oligo GC%\n\nThe [Inside Target Penalty](#PRIMER_INSIDE_PENALTY) and [Outside Target Penalty](#PRIMER_OUTSIDE_PENALTY) are similar, except that since they relate to position they do not lend them selves to the 'Lt' and 'Gt' nomenclature.\n\nFor the remaining parameters the optimum is understood and the actual value can only vary in one direction from the optimum:\n- Primer Self Complementarity\n- Primer 3' Self Complementarity\n- Primer #N's\n- Primer Mispriming Similarity\n- Primer Sequence Quality\n- Primer 3' Sequence Quality\n- Primer 3' Stability\n- Hyb Oligo Self Complementarity\n- Hyb Oligo 3' Self Complementarity\n- Hyb Oligo Mispriming Similarity\n- Hyb Oligo Sequence Quality\n- Hyb Oligo 3' Sequence Quality\n\nThe following are weights are treated specially:\n- Position Penalty Weight: Determines the overall weight of the position penalty in calculating the penalty for a primer.\n- Primer Weight: Determines the weight of the 2 primer penalties in calculating the primer pair penalty.\n- Hyb Oligo Weight: Determines the weight of the hyb oligo penalty in calculating the penalty of a primer pair plus hyb oligo.\n\nThe following govern the weight given to various parameters of primer pairs (or primer pairs plus hyb oligo).\n- T$_m$ difference\n- Primer-Primer Complementarity\n- Primer-Primer 3' Complementarity\n- Primer Pair Mispriming Similarit."}},
    {"header": {"body": "Hyb Oligos (Internal Oligos)", "anchor": "internal_oligo_generic"}},
    {"markdown": {"body": "Parameters governing choice of internal oligos are analogous to the parameters governing choice of primer pairs. The exception is Max 3' Complementarity which is meaningless when applied to internal oligos used for hybridization-based detection, since primer-dimer will not occur. We recommend that Max 3' Complementarity be set at least as high as Max Complementarity."}},
]

st_about = [
    {"header": {"body": "Primer3Plus"}},
    {"subheader": {"body": "Primer3Plus - Download Primer3Plus Program and Source Code"}},
    {"markdown": {"body": "[Source code is available at http://sourceforge.net/projects/primer3/.](http://sourceforge.net/projects/primer3/)"}},
    {"subheader": {"body": "Primer3Plus - Copyright Notice and Disclaimer"}},
    {"markdown": {"body": "Copyright (c) 2006, 2007\nby Andreas Untergasser and Harm Nijveen\nAll rights reserved.\n\nThe Primer3Plus is free software; you can redistribute it and/or modify\nit under the terms of the GNU General Public License as published by\nthe Free Software Foundation; either version 2 of the License, or\n(at your option) any later version.\n\nPrimer3Plus is distributed in the hope that it will be useful,\nbut WITHOUT ANY WARRANTY; without even the implied warranty of\nMERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the\nGNU General Public License for more details.\n\nTo received a copy of the GNU General Public License\nwrite to the Free Software Foundation, Inc., 51 Franklin St,\nFifth Floor, Boston, MA 02110-1301 USA"}},
    {"subheader": {"body": "Citing Primer3Plus"}},
    {"markdown": {"body": "Andreas Untergasser, Harm Nijveen, Xiangyu Rao, Ton Bisseling, René Geurts, and Jack A.M. Leunissen: **Primer3Plus, an enhanced web interface to Primer3** Nucleic Acids Research 2007 35: W71-W74; doi:10.1093/nar/gkm306"}},
    {"subheader": {"body": "Acknowledgments of Primer3Plus"}},
    {"markdown": {"body": "We thank Gerben Bijl for extensive beta-testing."}},
    {"header": {"body": "Primer3"}},
    {"subheader": {"body": "Primer3 - Download Primer3 Program and Source Code"}},
    {"markdown": {"body": "[Source code available at http://sourceforge.net/projects/primer3/.](http://sourceforge.net/projects/primer3/)"}},
    {"subheader": {"body": "Primer3 - Copyright Notice and Disclaimer"}},
    {"markdown": {"body": "Copyright (c) 1996,1997,1998,1999,2000,2001,2004,2006\nWhitehead Institute for Biomedical Research, [Steve Rozen](http://jura.wi.mit.edu/rozen/), and Helen Skaletsky\nAll rights reserved.\n\n```\nRedistribution and use in source and binary forms, with or without\nmodification, are permitted provided that the following conditions are\nmet:\n\n   * Redistributions of source code must retain the above copyright\nnotice, this list of conditions and the following disclaimer.\n   * Redistributions in binary form must reproduce the above\ncopyright notice, this list of conditions and the following disclaimer\nin the documentation and/or other materials provided with the\ndistribution.\n   * Neither the names of the copyright holders nor contributors may\nbe used to endorse or promote products derived from this software\nwithout specific prior written permission.\n\nTHIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS\n\"AS IS\" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT\nLIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR\nA PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT\nOWNERS OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,\nSPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT\nLIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,\nDATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY\nTHEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT\n(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE\nOF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.\n```"}},
    {"subheader": {"body": "Citing Primer3"}},
    {"markdown": {"body": "We request that use of this software be cited in publications as\n\n[Steve Rozen](http://jura.wi.mit.edu/rozen/) and Helen J. Skaletsky (2000) [Primer3 on the WWW for general users and for biologist programmers.](http://jura.wi.mit.edu/rozen/papers/rozen-and-skaletsky-2000-primer3.pdf) In: Krawetz S, Misener S (eds) *Bioinformatics Methods and Protocols: Methods in Molecular Biology.* Humana Press, Totowa, NJ, pp 365-386"}},
    {"subheader": {"body": "Acknowledgments of Primer3"}},
    {"markdown": {"body": "The development of Primer3 and the Primer3 web site was funded by Howard Hughes Medical Institute and by the National Institutes of Health, [National Human Genome Research Institute.](http://www.nhgri.nih.gov/) under grants R01-HG00257 (to David C. Page) and P50-HG00098 (to Eric S. Lander).\n\nWe thank [Centerline Software, Inc.,](http://www.centerline.com) for use of their TestCenter memory-error, -leak, and test-coverage checker.\n\nPrimer3 was a complete re-implementation of an earlier program: Primer 0.5 (*Steve Lincoln, Mark Daly, and Eric S. Lander*). *Lincoln Stein* championed the idea of making Primer3 a software component suitable for high-throughput primer design."}},
]
